src/test/java/com/teragrep/cfe_39/CombinedFullTest.java

```diff
@@ -0,0 +1,309 @@
+            Assertions.fail();
+        }
+        // Create a HDFS miniCluster
+        baseDir = Files.createTempDirectory("test_hdfs").toFile().getAbsoluteFile();
+        Configuration conf = new Configuration();
+        conf.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR, baseDir.getAbsolutePath());
+        MiniDFSCluster.Builder builder = new MiniDFSCluster.Builder(conf);
+        hdfsCluster = builder.build();
+        String hdfsURI = "hdfs://localhost:"+ hdfsCluster.getNameNodePort() + "/";
+        // System.out.println("hdfsURI: " + hdfsURI);
+        config.setHdfsuri(hdfsURI);
+        DistributedFileSystem fileSystem = hdfsCluster.getFileSystem();
+    }
+
+    // Teardown the minicluster
+    @AfterAll
+    public static void teardownMiniCluster() {
+        hdfsCluster.shutdown();
+        FileUtil.fullyDelete(baseDir);
+    }
+
+    @Test
+    public void kafkaAndAvroFullTest() throws InterruptedException, IOException {
+        insertMockFiles(); // Maybe add modifier that allows adding or not adding the mock file.
+        config.setMaximumFileSize(3000); // 10 loops (140 records) are in use at the moment, and that is sized at 36,102 bytes.
+        KafkaController kafkaController = new KafkaController(config);
+        Thread.sleep(10000);
+        kafkaController.run();
+        // The avro files should be committed to HDFS now. Check the committed files for any errors.
+        // There should be 20 files, 10 partitions with each having 2 files assigned to them.
+        // hdfsReadCheck(); does not work properly if pruning is enabled and prune offset is set too low, which causes the records to be pruned from the database.
+        if (config.getPruneOffset() == 157784760000L) {
+            try {
+                hdfsReadCheck();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+    }
+    @Test
+    public void hdfsPruneTest() throws IOException {
+        // Check that the files were properly written to HDFS with a read test.
+        String hdfsuri = config.getHdfsuri();
+
+        String path = config.getHdfsPath()+"/"+"testConsumerTopic";
+        // ====== Init HDFS File System Object
+        Configuration conf = new Configuration();
+        // Set FileSystem URI
+        conf.set("fs.defaultFS", hdfsuri);
+        // Because of Maven
+        conf.set("fs.hdfs.impl", org.apache.hadoop.hdfs.DistributedFileSystem.class.getName());
+        conf.set("fs.file.impl", org.apache.hadoop.fs.LocalFileSystem.class.getName());
+        // Set HADOOP user
+        System.setProperty("HADOOP_USER_NAME", "hdfs");
+        System.setProperty("hadoop.home.dir", "/");
+        //Get the filesystem - HDFS
+        FileSystem fs = FileSystem.get(URI.create(hdfsuri), conf);
+
+        //==== Create directory if not exists
+        Path workingDir=fs.getWorkingDirectory();
+        Path newDirectoryPath= new Path(path);
+        if(!fs.exists(newDirectoryPath)) {
```

Comment:
Tests should start from known state, at most data should be cleaned/prepared at startup, not conditionally mid test

Reply:
Solved in commit 65342c2e865912f981dfe82f470cab6884f72cdb, 95d01d59f793672aa22c8973d2328f249c240762 and dee838a1920ce47e53594859188692fde46535f7.